both/methods/read/channels.js

```diff
@@ -0,0 +1,35 @@
+Meteor.methods({
+  /*
+   * Reads a Channel and searches for its images
+   *
+   * @param {Channel} the Channel object to read
+   * @return {Images} an array of image objects
+   */
+  '/channels/': function(channel) {
+    // Check Arguments
+    check(channel, Channel);
+
+    // Check if the user is logged in? Not sure this is necessary
+    // if(!Meteor.user()) throw new Meteor.Error('logged-out','You must be logged in to create a channel.');
+    
+    if (channel.validate()) {
+      return Modules.both.getImages(channel);
+    } else {
+      channel.throwValidationsException();
+    }
+  },
+  
+  getUrls: function(channel) {
+    // Check Arguments
+    check(channel, Channel);
+
+    // Check if the user is logged in? Not sure this is necessary
+    // if(!Meteor.user()) throw new Meteor.Error('logged-out','You must be logged in to create a channel.');
+    
+    if (channel.validate()) {
+      return Modules.both.getUrls(channel);
+    } else {
+      channel.throwValidationsException();
+    }
+  }
+});
```

Comment:
Yeah so from the client, the astronomy objects gets converted to JSON, and
then on the way back, it gets converted back into an Astronomy object. So
passing them into method calls is no problem.

On Wednesday, September 30, 2015, Hunter Martin <notifications@github.com>
wrote:

> In both/methods/read/channels.js
> <https://github.com/elGatoMantocko/BetterBackground/pull/7#discussion_r40829707>
> :
>
> > +  },
> > +
> > +  getUrls: function(channel) {
> > +    // Check Arguments
> > +    check(channel, Channel);
> > +
> > +    // Check if the user is logged in? Not sure this is necessary
> > +    // if(!Meteor.user()) throw new Meteor.Error('logged-out','You must be logged in to create a channel.');
> > +
> > +    if (channel.validate()) {
> > +      return Modules.both.getUrls(channel);
> > +    } else {
> > +      channel.throwValidationsException();
> > +    }
> > +  }
> > +});
>
> What about coming from the client? Like if the client needs the urls for a
> channel, they need to send the channel data over DDP and this method is
> expecting an Astronomy object. Does that mean they can just format a JSON
> object and it will automatically create an Astronomy object?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elGatoMantocko/BetterBackground/pull/7/files#r40829707>
> .
>